snapcraft/storeapi/v2/releases.py

```diff
@@ -0,0 +1,180 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Any, Dict, List
+
+import jsonschema
+
+from ._api_schema import RELEASES_JSONSCHEMA
+
+"""
+This module holds representations for results for the v2 releases
+API endpoint provided by the Snap Store. It currently only covers
+Revisions and Releases.
+
+The full API is documented on
+https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-releases
+"""
+
+
+class Release:
+    """Represent a snap release."""
+
+    @classmethod
+    def unmarshal(cls, payload: Dict[str, Any]) -> "Release":
+        jsonschema.validate(
+            payload,
+            RELEASES_JSONSCHEMA["properties"]["releases"]["items"]["properties"],
+        )
+        return cls(
+            architecture=payload["architecture"],
+            branch=payload["branch"],
+            channel=payload["channel"],
+            expiration_date=payload["expiration-date"],
+            revision=payload["revision"],
+            risk=payload["risk"],
+            track=payload["track"],
+            when=payload["when"],
+        )
+
+    def marshal(self) -> Dict[str, Any]:
+        return {
+            "architecture": self.architecture,
+            "branch": self.branch,
+            "channel": self.channel,
+            "expiration-date": self.expiration_date,
+            "revision": self.revision,
+            "risk": self.risk,
+            "track": self.track,
+            "when": self.when,
+        }
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}: {self.channel!r}>"
+
+    def __init__(
+        self,
+        architecture: str,
+        branch: str,
+        channel: str,
+        expiration_date: str,
+        revision: int,
```

Comment:
revision is Optional according to spec
```suggestion
        revision: Optional[int],
```

Reply:
this one might be a mistake (if using the jsonschema as an entry point), a release requires a revision

Reply:
I would agree with that in practice, I just think if the schema you're using for validation allows it, it should asserted when unmarshalling.

Reply:
I will remove it from the jsonschema then

Reply:
schema has been fixed